packages/@react-spectrum/link/stories/Link.stories.tsx

```diff
@@ -10,62 +10,77 @@
-  )
-  .add(
-    'variant: overBackground',
-    () => (
+type LinkStory = ComponentStoryObj<typeof Link>;
+
+export default {
+  title: 'Link',
+  component: Link,
+  argTypes: {
+    onPress: {
+      action: 'press'
+    },
+    onPressStart: {
+      action: 'pressstart'
+    },
+    onPressEnd: {
+      action: 'pressend'
+    }
+  }
+} as ComponentMeta<typeof Link>;
+
+export let Default: LinkStory = {
+  args: {children: 'This is a React Spectrum Link'}
+};
+
+export let Secondary: LinkStory = {
+  ...Default,
+  args: {...Default.args, variant: 'secondary'},
+  name: 'variant: secondary'
+};
+
+export let OverBackground: LinkStory = {
+  ...Default,
+  args: {...Default.args, variant: 'overBackground'},
+  decorators: [
+    (Story) => (
       <div style={{backgroundColor: 'rgb(15, 121, 125)', color: 'rgb(15, 121, 125)', padding: '15px 20px', display: 'inline-block'}}>
-        {render({variant: 'overBackground', onPress: action('press'), onPressStart: action('pressstart'), onPressEnd: action('pressend')})}
+        <Story />
       </div>
     )
-  )
-  .add(
-    'isQuiet: true',
-    () => render({isQuiet: true, onPress: action('press'), onPressStart: action('pressstart'), onPressEnd: action('pressend')})
-  )
-  .add(
-    'isQuiet: true, variant: secondary',
-    () => render({isQuiet: true, variant: 'secondary', onPress: action('press'), onPressStart: action('pressstart'), onPressEnd: action('pressend')})
-  )
-  .add(
-    'children: a',
-    () => renderWithChildren({onPress: action('press'), onPressStart: action('pressstart'), onPressEnd: action('pressend')})
-  )
-  .add(
-    'onPress',
-    () => render({onPress: action('press'), onPressStart: action('pressstart'), onPressEnd: action('pressend')})
-  )
-  .add(
-    'onClick',
-    () => render({onClick: action('deprecatedOnClick'), onPress: action('press'), onPressStart: action('pressstart'), onPressEnd: action('pressend')})
-  );
+  ],
+  name: 'variant: overBackground'
+};
+
+export let IsQuiet: LinkStory = {
+  ...Default,
+  args: {...Default.args, isQuiet: true},
+  name: 'isQuiet: true'
+};
+
+export let IsQuietSecondary: LinkStory = {
+  ...Default,
+  args: {...IsQuiet.args, ...Secondary.args},
+  name: 'isQuiet: true, variant: secondary'
+};
 
-function render(props = {}) {
-  return (
-    <Link {...props}>
-      This is a React Spectrum Link
-    </Link>
-  );
-}
+export let WithChildren: LinkStory = {
+  ...Default,
+  args: {children: <a href="//example.com" target="_self">This is a React Spectrum Link</a>},
+  name: 'children: a'
+};
 
-function renderWithChildren(props = {}) {
-  return (
-    <Link {...props}>
-      <a href="//example.com" target="_self">This is a React Spectrum Link</a>
-    </Link>
-  );
-}
+export let OnClick: LinkStory = {
+  ...Default,
+  args: {...Default.args},
+  argTypes: {
+    onClick: {
+      action: 'deprecatedOnClick'
+    }
+  } as Partial<ArgTypes<SpectrumLinkProps>> & {onClick: any},
```

Comment:
Weird case since onClick isn't in the types. Let me know if there is a better way to handle

Reply:
yep, it's ts-ignored, i think this is fine for this story